### PR TITLE
PDF editor: Upgrade fabric.js

### PR DESCRIPTION
**Pull request by raphaelm**
- [x] Fix text colors
- [x] Fix group handling
- [x] Test everything

### CODE REVIEW

**Review by ser8phin — APPROVED** [review]
Looks good, editor works as expected :+1:

**Comment by raphaelm** [comment]
Thanks. Will not merge today as I'm not in the office tomorrow, but later this week.

File: src/pretix/static/pretixcontrol/js/ui/editor.js [hunk 1]
```diff
@@ -397,6 +398,7 @@ var editor = {
         editor.fabric.on('object:added', editor._create_savepoint);
         editor.fabric.on('selection:cleared', editor._update_toolbox);
         editor.fabric.on('selection:created', editor._update_toolbox);
+        editor.fabric.on('selection:updated', editor._update_toolbox);
         editor.fabric.on('object:selected', editor._update_toolbox);
```
Comment (wiffbi) [inline src/pretix/static/pretixcontrol/js/ui/editor.js:402]: From the [changelog](http://fabricjs.com/v2-breaking-changes-2) object:selected was deprecated in v2 and removed in v3. So we could (should?) remove it. I removed it and thanks to selection:created it still works.

```suggestion
```

File: src/pretix/static/pretixcontrol/js/ui/editor.js [hunk 2]
```diff
@@ -397,6 +398,7 @@ var editor = {
...
         editor.fabric.on('object:moving', editor._update_toolbox_values);
```
Comment (wiffbi) [inline src/pretix/static/pretixcontrol/js/ui/editor.js:403]: As noted in the [changelog](http://fabricjs.com/v5-breaking-changes) for v5, `object:moving, object:rotating` and `object:scaling` are just an „extra layer detail on top of `object:modified`“. As all of these events call `editor._update_toolbox` all `editor.fabric.on('object:XXX',` except `editor.fabric.on('object:modified',` could/should be removed. See lines 405 and 406 as well – could not select them in the UI here.

```suggestion
```

**Pull request merged**